client/crouchprone.lua

```diff
@@ -31,71 +32,81 @@ RegisterNetEvent('crouchprone:client:SetWalkSet', function(clipset)
     walkSet = clipset
 end)
 
-CreateThread(function()
-    local sleep
-    while true do
-        sleep = 1000
-        local ped = PlayerPedId()
-        if not IsPedSittingInAnyVehicle(ped) and not IsPedFalling(ped) and not IsPedSwimming(ped) and not IsPedSwimmingUnderWater(ped) then
-            sleep = 0
-            if IsControlJustReleased(2, 36) then
-                stage += 1
+function crouch()
```

Comment:
I don't believe this is going to work by being a function like this.

If I understand your implementation correctly, when you press control, it calls the function `crouch()`.
Which means everytime I press control it will run the function and create a new thread each time preventing you from ever reaching stages 2 or 3?

I can't immediately think of a way around it. Happy to discuss if I you think I have got this wrong.